contents/java/Java.md

```diff
@@ -14,6 +14,161 @@ Authors: John Doe, Jane Doe
 
 ## Reflection
 
-// Reflection overview
+### What is Java reflections
+
+Java reflections allow the inspection of classes, methods and fields during runtime without having any knowledge of it
+during compile time. It is also possible to new instance of existing objects, invoke methods and changing values of
+fields. Hence the name reflection because Java is in a way looking at itself during runtime!
+
+### The Basics of Reflections
+
+Before getting started with reflections in Java, it is important to realise that classes itself is an object. Every
+unique `Object` is assigned an immutable `Class` object by the JVM. This is fundamentally different from *instances* of
```

Comment:
You may want to replace 'This' with 'This immutable `Class` object' as I think it may not be clear what 'This' is referring to. (Is is the fact that every object is assigned a class object? Is it the unique object? Or is it the class object?)